ast/src/parsed/mod.rs

```diff
@@ -42,23 +42,24 @@ pub enum SymbolCategory {
     TypeConstructor,
     /// A trait declaration, which can be used as a type.
```

Comment:
Is this implemented like that? I actually don't think we currently do that, and for simplicity, I would prefer to keep it like `let<T: MyTrait<T>> f: T -> int` for now. We can introduce a shorthand like `let f: MyTrait -> int` later like rust does it.